modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -3,22 +3,22 @@ import {GeoJsonLayer} from '@deck.gl/layers';
 import TileCache from './utils/tile-cache';
 
 const defaultProps = {
-  renderSubLayers: props => new GeoJsonLayer(props),
-  getTileData: ({x, y, z}) => Promise.resolve(null),
-  onDataLoaded: () => {},
+  renderSubLayers: {type: 'function', value: props => new GeoJsonLayer(props)},
+  getTileData: {type: 'function', value: ({x, y, z}) => Promise.resolve(null)},
+  onTileLoaded: {type: 'function', value: () => {}},
   // eslint-disable-next-line
-  onGetTileDataError: err => console.error(err),
-  maxZoom: null,
-  minZoom: null,
-  maxCacheSize: null
+  onTileError: {type: 'function', value: err => console.error(err)},
+  maxZoom: {type: 'number', min: 0, value: null},
+  minZoom: {type: 'number', min: 0, value: null},
```

Comment:
- `null` will throw an error when checking against type `number`. It's ok to not specify the type of a prop.
- Default `minZoom` should be 0.